presto-main/src/main/java/com/facebook/presto/sql/gen/SwitchCodeGenerator.java

```diff
@@ -97,49 +81,84 @@ else if (<value> == <non-constant2>) {
         // determine the type of the value and result
         Class<?> valueType = value.getType().getJavaType();
 
+        // We generate SearchedCase as CASE TRUE WHEN p1 THEN v1 WHEN p2 THEN p2...
+        boolean searchedCase = (value instanceof ConstantExpression && ((ConstantExpression) value).getType() == BOOLEAN &&
+                ((ConstantExpression) value).getValue() == Boolean.TRUE);
+
         // evaluate the value and store it in a variable
-        LabelNode nullValue = new LabelNode("nullCondition");
-        Variable tempVariable = scope.createTempVariable(valueType);
-        BytecodeBlock block = new BytecodeBlock()
-                .append(valueBytecode)
-                .append(BytecodeUtils.ifWasNullClearPopAndGoto(scope, nullValue, void.class, valueType))
-                .putVariable(tempVariable);
-
-        BytecodeNode getTempVariableNode = VariableInstruction.loadVariable(tempVariable);
-
-        // build the statements
-        elseValue = new BytecodeBlock().visitLabel(nullValue).append(elseValue);
-        // reverse list because current if statement builder doesn't support if/else so we need to build the if statements bottom up
-        for (RowExpression clause : Lists.reverse(whenClauses)) {
+        LabelNode elseLabel = new LabelNode("else");
+        LabelNode endLabel = new LabelNode("end");
+
+        BytecodeBlock block = new BytecodeBlock();
+        BytecodeNode getTempVariableNode = null;
```

Comment:
We normally would use a `Optional<BytecodeNode>` in this situation rather than using null.

Reply:
Done